.github/workflows/ci-additional.yaml

```diff
@@ -152,6 +150,42 @@ jobs:
         run: |
           python -m pytest --doctest-modules xarray --ignore xarray/tests
 
+  mypy:
+    name: Mypy
+    runs-on: "ubuntu-latest"
+    if: needs.detect-ci-trigger.outputs.triggered == 'false'
+    defaults:
+      run:
+        shell: bash -l {0}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # Fetch all history for all branches and tags.
+      - uses: conda-incubator/setup-miniconda@v2
+        with:
+          channels: conda-forge
+          channel-priority: strict
+          mamba-version: "*"
+          activate-environment: xarray-tests
+          auto-update-conda: false
+          python-version: "3.9"
+
+      - name: Install conda dependencies
+        run: |
+          mamba env update -f ci/requirements/environment.yml
+      - name: Install xarray
+        run: |
+          python -m pip install --no-deps -e .
+      - name: Version info
+        run: |
+          conda info -a
+          conda list
+          python xarray/util/print_versions.py
```

Comment:
This seems like a lot of boilerplate. Am I doing it correctly?

Reply:
Potentially we could turn this setup into an Action; we did this in prql quite successfully: https://github.com/prql/prql/blob/main/.github/actions/cargo-test/action.yaml